src/app/(marketing)/profile/[username]/page.tsx

```diff
@@ -0,0 +1,195 @@
+import { CheckMark } from "@/components/CheckMark"
+import {
+  Layout,
+  LayoutBanner,
+  LayoutFooter,
+  LayoutHeader,
+  LayoutHero,
+  LayoutMain,
+} from "@/components/Layout"
+import { Anchor } from "@/components/ui/Anchor"
+import { Avatar, AvatarFallback, AvatarImage } from "@/components/ui/Avatar"
+import { AvatarGroup } from "@/components/ui/AvatarGroup"
+import { Badge } from "@/components/ui/Badge"
+import { Button } from "@/components/ui/Button"
+import { Card } from "@/components/ui/Card"
+import { Progress } from "@/components/ui/Progress"
+import { Separator } from "@/components/ui/Separator"
+import { Skeleton } from "@/components/ui/Skeleton"
+import { Schemas } from "@guildxyz/types"
+import { ArrowRight } from "@phosphor-icons/react/dist/ssr"
+import { ActivityChart } from "../_components/ActivityChart"
+import { CircularProgressBar } from "../_components/CircularProgressBar"
+import { ContributionCard } from "../_components/ContributionCard"
+import { EditContributions } from "../_components/EditContributions"
+import { EditProfile } from "../_components/EditProfile"
+import { LevelBadge } from "../_components/LevelBadge"
+import { OperatedGuildCard } from "../_components/OperatedGuildCard"
+import { RecentActivity } from "../_components/RecentActivity"
+
+async function getProfileData(username: string) {
+  const req = `https://api.guild.xyz/v2/profiles/${username}`
+  try {
+    const res = await fetch(req)
+    if (!res.ok) {
+      throw new Error("Failed to fetch profile data")
+    }
+    return res.json() as Promise<Schemas["Profile"]>
+  } catch (e) {
+    // mocking for the time being if fetch fails
+    console.error(e)
+    const res = {
+      id: 4,
+      userId: 6027190,
+      username: "durumm",
+      name: "durum",
+      bio: null,
+      profileImageUrl:
+        "https://guild-xyz.mypinata.cloud/ipfs/QmWGdo6FkjSz22oDZFMJysx3hGKoVqtzTWVMx9tTSP7jvi",
+      backgroundImageUrl: null,
+      createdAt: "2024-07-25T10:04:20.781Z",
+      updatedAt: "2024-07-25T10:04:20.781Z",
+    }
+    return res
+  }
+}
+
+const Page = async ({
+  params: { username },
+}: {
+  params: { username: string }
+}) => {
+  const profile = await getProfileData(username)
+  const level = 50
+
+  return (
+    <Layout>
+      <LayoutHero>
+        <LayoutHeader />
+        <LayoutBanner className="-bottom-[500px]">
+          <div className="absolute inset-0 bg-[url('/banner.svg')] opacity-10" />
+          <div className="absolute inset-0 bg-gradient-to-t from-background" />
+        </LayoutBanner>
+      </LayoutHero>
+      <LayoutMain>
+        <div className="mt-24">
+          <div className="relative mb-24 flex flex-col items-center">
+            <EditProfile
+              profileImageUrl={profile.profileImageUrl}
```

Comment:
What if we'd wrap the whole page in `SWRConfig` & provide the server side fetched data to it as a default value & create a `useProfile` hook to retrieve the necessary data inside each component? That way we wouldn't need to pass so many parameters here and in other places.